src/OSPSuite.Core/Domain/PKAnalyses/DynamicPKParameter.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Linq;
+using OSPSuite.Core.Domain.Services;
+
+namespace OSPSuite.Core.Domain.PKAnalyses
+{
+   public enum StandardPKParameter
+   {
+      Unknown,
+      Cmax,
+      Cmin,
+      Tmax,
+      Tmin,
+      CTrough,
+      Auc,
+      Aucm,
+      AucInf,
+      AucTendInf,
+      Mrt,
+      FractionAucEndToInf,
+      Thalf,
+      Vss,
+      Vd,
+      Tthreshold
+   }
+
+   public class DynamicPKParameter : PKParameter
+   {
+      // Parameter such as AUC, Cmax, Tmax that will be used to calculate the parameter
+      public StandardPKParameter StandardPKParameter { get; set; } = StandardPKParameter.Unknown;
+
+      public float? StartTime { get; set; }
+      public float? EndTime { get; set; }
+
+      /// <summary>
+      ///    1-based index of the application that should be used to estimate the start of the interval
+      /// </summary>
+      public int? StartApplication { get; set; }
+
+      /// <summary>
+      ///    1-based index of the application that should be used to estimate the end of the interval
+      /// </summary>
+      public int? EndApplication { get; set; }
+
+      /// <summary>
+      ///    Time offset in min to apply to the start time (either from the absolute <see cref="StartTime" /> or the start time
+      ///    estimated from the application <see cref="StartApplication" />
+      /// </summary>
+      public float? StartTimeOffset { get; set; }
+
+      /// <summary>
+      ///    Time offset in min to apply to the start time (either from the absolute <see cref="EndTime" /> or the start time
+      ///    estimated from the application <see cref="EndApplication" />
+      /// </summary>
+      public float? EndTimeOffset { get; set; }
+      
+      /// <summary>
+      /// Dose parameter that will be used to calculate a normalized parameter. Not that if specified, only the normalized parameter will be added
+      /// </summary>
+      public float? DoseForNormalization { get; set; }
```

Comment:
what about the **unit** of `DoseForNormalization`? Or is it supposed to be always in the **internal** unit and it's responsibility of the caller?

Reply:
Yes absolutely. If we have the unit, what are we going to do with it? Then we need to manage the dimension as well etc. 
When using something like this, the user will have an object in unit "[BaseParameter]/[Normalization]

Reply:
And it is called NormalizationFactor now :)